overlord/auth/auth.go

```diff
@@ -28,14 +28,28 @@ import (
 
 	"gopkg.in/macaroon.v1"
 
+	"github.com/snapcore/snapd/asserts"
 	"github.com/snapcore/snapd/overlord/state"
 	"github.com/snapcore/snapd/store"
 )
 
 // AuthState represents current authenticated users as tracked in state
 type AuthState struct {
-	LastID int         `json:"last-id"`
-	Users  []UserState `json:"users"`
+	LastID int          `json:"last-id"`
+	Users  []UserState  `json:"users"`
+	Device *DeviceState `json:"device,omitempty"`
+}
+
+// DeviceState represents the device's identity and store credentials
+type DeviceState struct {
+	Brand  string `json:"brand,omitempty"`
+	Model  string `json:"model,omitempty"`
+	Serial string `json:"serial,omitempty"`
+	// XXX: SerialAssertion should eventually be retrieved from the
+	// assertions DB, not persisted in state.
```

Comment:
None of this information should be kept out of the assertion, I believe, otherwise when should we use it?  When is it okay to consider correct information that we did not validate or checked for consistency?